test/e2e/multiple_instances_test.go

```diff
@@ -0,0 +1,99 @@
+// +build multiple
+
+package e2e
+
+import (
+	"context"
+	"testing"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type MulitpleInstanceTestSuite struct {
+	suite.Suite
+}
+
+func (suite *MulitpleInstanceTestSuite) SetupSuite() {
+	t = suite.T()
+	var err error
+	ctx, err = prepare(t)
+	if err != nil {
+		if ctx != nil {
+			ctx.Cleanup()
+		}
+		require.FailNow(t, "Failed in prepare")
+	}
+	fw = framework.Global
+	namespace = ctx.GetID()
+	require.NotNil(t, namespace, "GetID failed")
+
+	addToFrameworkSchemeForSmokeTests(t)
+}
+
+func (suite *MulitpleInstanceTestSuite) TearDownSuite() {
+	handleSuiteTearDown()
+}
+
+func TestMultipleInstanceSuite(t *testing.T) {
+	suite.Run(t, new(MulitpleInstanceTestSuite))
+}
+
+func (suite *MulitpleInstanceTestSuite) SetupTest() {
+	t = suite.T()
+}
+
+func (suite *MulitpleInstanceTestSuite) AfterTest(suiteName, testName string) {
+	handleTestFailure()
+}
+
+/*
+ * This test verifies that we create the elasticsearch secrets correctly if someone creates production Jaeger
+ * instances with the same name in different namespaces
+ */
+func (suite *MulitpleInstanceTestSuite) TestVerifySecrets() {
```

Comment:
Stupid question, but are we sure this test fails when the fix isn't applied?

Reply:
@jpkrohling Yes, I ran it on OCP 4.5 with the 1.17.6 Jaeger Operator and it failed.